test/services/optIn/SignUpSubmissionServiceSpec.scala

```diff
@@ -0,0 +1,193 @@
+
+  val service: SignUpSubmissionService =
+    new SignUpSubmissionService(mockAuditingService, mockItsaStatusUpdateConnector, mockOptInService, mockUiJourneySessionDataRepository)
+
+  "SignUpUpdateService" when {
+
+    ".getOptInSessionData()" when {
+
+      "there is data pre-existing in db" should {
+
+        "return the correct session data" in {
+
+          val optInContextData =
+            OptInContextData(
+              currentTaxYear = "2025-2026",
+              currentYearITSAStatus = "MTD Voluntary",
+              nextYearITSAStatus = "MTD Voluntary"
+            )
+
+          val selectedOptInYear = Some("2025-2026")
+          val optInSessionData = OptInSessionData(Some(optInContextData), selectedOptInYear = selectedOptInYear)
+
+          val retrievedUiSessionData =
+            UIJourneySessionData(
+              sessionId = hc.sessionId.get.value,
+              journeyType = Opt(OptInJourney).toString,
+              optInSessionData = Some(optInSessionData)
+            )
+
+          when(mockUiJourneySessionDataRepository.get(hc.sessionId.get.value, Opt(OptInJourney)))
+            .thenReturn(Future(Some(retrievedUiSessionData)))
+
+          val actual = service.getOptInSessionData()
+          whenReady(actual) { result => result shouldBe Some(optInSessionData) }
+        }
+      }
+
+      "there is NO data present in db" should {
+
+        "return None" in {
+
+          when(mockUiJourneySessionDataRepository.get(hc.sessionId.get.value, Opt(OptInJourney)))
+            .thenReturn(Future(None))
+
+          val actual = service.getOptInSessionData()
+          whenReady(actual) { result => result shouldBe None }
+        }
+      }
+    }
+
+    ".triggerOptInRequest()" when {
+
+      "the user has selected a tax year" should {
+
+        "successfully make a Sign up request" in {
+
+          val optInContextData =
+            OptInContextData(
+              currentTaxYear = "2025-2026",
+              currentYearITSAStatus = "MTD Voluntary",
+              nextYearITSAStatus = "MTD Voluntary"
+            )
+
+          val selectedOptInYear = Some("2025-2026")
+          val optInSessionData = OptInSessionData(Some(optInContextData), selectedOptInYear = selectedOptInYear)
+
+          val retrievedUiSessionData =
+            UIJourneySessionData(
+              sessionId = hc.sessionId.get.value,
+              journeyType = Opt(OptInJourney).toString,
+              optInSessionData = Some(optInSessionData)
+            )
+
+          val currentOptInTaxYear =
+            CurrentOptInTaxYear(Voluntary, TaxYear(2025, 2026))
+
+          val nextOptInTaxYear =
+            NextOptInTaxYear(Voluntary, TaxYear(2026, 2027), currentOptInTaxYear)
+
+          val optInProposition =
+            OptInProposition(currentOptInTaxYear, nextOptInTaxYear)
+
+          when(mockUiJourneySessionDataRepository.get(hc.sessionId.get.value, Opt(OptInJourney)))
```

Comment:
There's no tests/functionality for multi year

Reply:
added some tests and changed some of the logic and validation to handle single and multi years 🤞🏼